### PR TITLE
`Gathering data`: hide ligand names flag

### LINKED ISSUE #190: Gathering: replace ligand names with random strings
Some of our partners have run the private datasets with internal ligand names. We will need to replace the ligand names in the gathering script and also provide a table of private to public names for the partners.

### PR DESCRIPTION
**Pull request by jthorton**
## Summary of changes

Fixes #190 by adding a flag to change the ligand names.

## Licensing agreement
* [x] I agree to release these inputs under the combined MIT and CCBY SA 4.0 licenses of this repository

[templates]: https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/tree/main/industry_benchmarks
[input_validation]: https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/tree/main/industry_benchmarks/utils/input_validation.py

### CODE REVIEW

**Review by IAlibay — CHANGES_REQUESTED** [review]
Looks good, couple of comments

File: industry_benchmarks/utils/data_gathering.py [hunk 1]
```diff
@@ -505,6 +552,9 @@ def gather_data(
         "transformation_scores": transformation_scores,
         "ligand_scores": ligand_scores,
     }
+    if hide_ligand_names:
```
[the remark this answers is not in the document]
Comment (jthorton) [reply industry_benchmarks/utils/data_gathering.py:555]: Ahh yes good point!

File: industry_benchmarks/utils/data_gathering.py [hunk 2]
```diff
@@ -496,6 +540,18 @@ def gather_data(
         fixed_network = parse_ligand_network(fixed_ligand_network)
         ligand_network = ligand_network.enlarge_graph(
             edges=fixed_network.edges, nodes=fixed_network.nodes)
+
+    if hide_ligand_names:
+        ligand_network, name_mapping = replace_ligand_names(ligand_network)
+        with open(output_dir / "ligand_name_mapping.json", "w") as out:
```
Comment (hannahbaumann) [inline industry_benchmarks/utils/data_gathering.py:546]: I forgot, will we in the end zip up this whole output directory and ask them to send it to us or should they send the individual files within the directory? If the former, this file should be stored elsewhere.

**Review by hannahbaumann — COMMENTED** [review]
Thanks @jthorton , LGTM, just left that one comment!

File: industry_benchmarks/utils/data_gathering.py [hunk 3]
```diff
@@ -496,6 +540,18 @@ def gather_data(
...
+            json.dump(name_mapping, out)
+    else:
+        # make some noise that we are not hiding the names
+        import warnings
+        warnings.warn(message="The names of the ligands will be used in the results, if you want to make them anonymous "
```
Comment (IAlibay) [inline industry_benchmarks/utils/data_gathering.py:551]:
```suggestion
        warnings.warn(message="The names of the ligands will be used as-is in the results, if you want to make them anonymous "
```
Or something like that - the wording just seemed a tiny bit unclear sorry.

File: industry_benchmarks/utils/data_gathering.py (hunk 2 above)
Comment (IAlibay) [reply industry_benchmarks/utils/data_gathering.py:546]: Ah yeah that's a good point, folks are 100% going to forget to remove the file if it ends up in the same directory.

Maybe it needs to be nested:

output_dir/SENDTHISDIRECTORYTOUS
output_dir/somefilesforyou

**Review by IAlibay — CHANGES_REQUESTED** [review]

**Comment by IAlibay** [comment]
P.S. Please feel free to discard my review and merge when things are suitably sorted out - it's probably best not to spend a ton of time cleaning up the file writing here if we're going to do it elsewhere.

**Review by IAlibay — APPROVED** [review]
approving early so I'm not blocking

**Review by hannahbaumann — APPROVED** [review]
Thanks @jthorton , lgtm. We can address the output file structure in a separate PR if needed!

**Review by hannahbaumann — APPROVED** [review]
Thanks @jthorton , lgtm!

**Pull request merged**